omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/AlertRecipientResource2_0.java

```diff
@@ -0,0 +1,190 @@
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs2_0;
+
+import io.swagger.models.Model;
+import io.swagger.models.ModelImpl;
+import io.swagger.models.properties.BooleanProperty;
+import io.swagger.models.properties.DateTimeProperty;
+import io.swagger.models.properties.RefProperty;
+import io.swagger.models.properties.StringProperty;
+import org.apache.commons.lang.StringUtils;
+import org.openmrs.User;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.PropertyGetter;
+import org.openmrs.module.webservices.rest.web.annotation.PropertySetter;
+import org.openmrs.module.webservices.rest.web.annotation.SubResource;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingSubResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.openmrs.notification.Alert;
+import org.openmrs.notification.AlertRecipient;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Sub-resource for {@link AlertRecipient}
+ */
+@SubResource(parent = AlertResource2_0.class, path = "recipient", supportedClass = AlertRecipient.class, supportedOpenmrsVersions = {
+		"2.0.*", "2.1.*", "2.2.*", "2.3.*", "2.4.*", "2.5.*" })
+public class AlertRecipientResource2_0 extends DelegatingSubResource<AlertRecipient, Alert, AlertResource2_0> {
+
+	private static final String RECIPIENT = "recipient";
+
+	private static final String ALERT_READ = "alertRead";
+
+	private static final String DATE_CHANGED = "dateChanged";
+
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("uuid");
+		description.addProperty("display");
+		description.addSelfLink();
+
+		if (rep instanceof DefaultRepresentation) {
+			description.addProperty(RECIPIENT, Representation.REF);
```

Comment:
I find these constants a bit different from what is else where.

Reply:
What do you mean? :)

Reply:
Do existing resources also use constants for the same?

Reply:
They don't but using constants for duplicate strings is a good practice.